sidekick_core/test/flutter_test.dart

```diff
@@ -4,27 +4,39 @@ import 'package:sidekick_core/sidekick_core.dart';
 import 'package:test/test.dart';
 
 void main() {
-  group('systemFlutterSdkPath', () {
-    test('returns null when flutter is not on path', () {
-      final whichPath =
-          start('which which', progress: Progress.capture()).firstLine!;
-      PATH.where((p) => !whichPath.startsWith(p)).forEach(env.removeFromPATH);
+  final tempDir = Directory.systemTemp.createTempSync();
+  final fakeFlutter = tempDir.file('fake_flutter/bin/flutter')
+    ..createSync(recursive: true);
+  chmod(fakeFlutter.path, permission: '755');
+
+  setUp(() {
+    final whichPath =
+        start('which which', progress: Progress.capture()).firstLine!;
+    PATH.where((p) => !whichPath.startsWith(p)).forEach(env.removeFromPATH);
+    env['FLUTTER_ROOT'] = null;
```

Comment:
Bit hacky but it works. 

dcli Env can be manipulated with a [scope](https://pub.dev/packages/scope), but I never got it working. https://github.com/onepub-dev/dcli/blob/master/dcli_core/lib/src/functions/env.dart#L86

see https://github.com/onepub-dev/dcli/blob/b50b8fb57809f850f15b32c53f22f3ad9b5c2197/dcli/test/src/util/test_scope.dart#L24

Reply:
I just tried it with a scope but couldn't get it to work either. I managed to change the env, however `which flutter` still worked and returned something even though I overrode `env` with an empty map (so `PATH` is empty too)